src/core/CCNativeint.ml

```diff
@@ -0,0 +1,51 @@
+(* This file is free software, part of containers. See file "license" for more details. *)
+
+include Nativeint
+
+let equal (x:t) y = Pervasives.(=) x y
+
+module Infix = struct
+  let (+) = add
+
+  let (-) = sub
+
+  let (~-) = neg
+
+  let ( * ) = mul
+
+  let (/) = div
+
+  let (mod) = rem
+
+  let (land) = logand
+
+  let (lor) = logor
+
+  let (lxor) = logxor
+
+  let lnot = lognot
+
+  let (lsl) = shift_left
+
+  let (lsr) = shift_right_logical
+
+  let (asr) = shift_right
+
+  let (=) = equal
+
+  let (<>) = Pervasives.(<>)
+  let (<) = Pervasives.(<)
+  let (<=) = Pervasives.(<=)
+  let (>) = Pervasives.(>)
+  let (>=) = Pervasives.(>=)
+end
+include Infix
+
+let hash x = Pervasives.abs (to_int x)
```

Comment:
This hash function does not give the same result as the standard ```Hashtbl.hash``` function.

@c-cube is that a problem ?

Reply:
It's not a problem, we never promised that. The hash only has to be deterministic (and hopefully not too far from uniformity)